test/box-luatest/session_test.lua

```diff
@@ -0,0 +1,311 @@
+    end)
+end
+
+g.test_check_session = function(cg)
+    cg.server:exec(function()
+        local fiber = require('fiber')
+        local session = box.session
+        t.assert_gt(session.id(), 0, "id > 0")
+        local failed = false
+        local f = fiber.create(function() failed = session.id() == 0 end)
+        while f:status() ~= 'dead' do fiber.sleep(0) end
+        t.assert(not failed, 'session not broken')
+        t.assert_equals(session.peer(), session.peer(session.id()),
+                        'peer() == peer(id())')
+    end)
+end
+
+g.test_check_session_triggers = function(cg)
+    cg.server:exec(function()
+        local session = box.session
+        local function
+            noop()
+        end
+
+        -- Check `on_connect/on_disconnect` triggers.
+        t.assert_type(session.on_connect(noop), 'function',
+                      'type of trigger noop on_connect')
+        t.assert_type(session.on_disconnect(noop), 'function',
+                      'type of trigger noop on_disconnect')
+
+        -- Check it's possible to reset these triggers.
+        local function fail() error('hear') end
+        t.assert_type(session.on_connect(fail, noop), 'function',
+                      'type of trigger fail, noop on_connect')
+        t.assert_type(session.on_disconnect(fail, noop), 'function',
+                      'type of trigger fail, noop on_disconnect')
+
+        -- Check on_connect/on_disconnect argument count and type.
+        t.assert_type(session.on_connect(), 'table',
+                      'type of trigger on_connect, no args')
+        t.assert_type(session.on_disconnect(), 'table',
+                      'type of trigger on_disconnect, no args')
+
+        local _, err = pcall(session.on_connect, function() end, function() end)
+        t.assert_str_contains(err, 'trigger reset: Trigger is not found',
+                              'on_connect trigger not found')
+        _, err = pcall(session.on_disconnect, function() end, function() end)
+        t.assert_str_contains(err, 'trigger reset: Trigger is not found',
+                              'on_disconnect trigger not found')
+
+        _, err = pcall(session.on_connect, 1, 2)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_connect bad args #1')
+        _, err = pcall(session.on_disconnect, 1, 2)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_disconnect bad args #1')
+
+        _, err = pcall(session.on_connect, 1)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_connect bad args #2')
+        _, err = pcall(session.on_disconnect, 1)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_disconnect bad args #2')
+
+        -- Use `nil` to clear the trigger.
```

Comment:
Minor: I suppose that this is not critical since we already have a clean-up of the server. But this also tests `session`'s functionality. So it is OK to leave it as is, plus add the corresponding comment. Here and below.